src/compute/src/logging/compute.rs

```diff
@@ -289,24 +285,28 @@ impl LirMetadata {
     }
 }
 
+pub(super) struct Return {
+    pub collections: BTreeMap<LogVariant, LogCollection>,
+}
+
 /// Constructs the logging dataflow for compute logs.
 ///
 /// Params
-/// * `worker`: The Timely worker hosting the log analysis dataflow.
+/// * `scope`: The Timely scope hosting the log analysis dataflow.
+/// * `scheduler`: The timely scheduler to obtainer activators.
 /// * `config`: Logging configuration.
 /// * `event_queue`: The source to read compute log events from.
-pub(super) fn construct<A: Allocate + 'static>(
-    worker: &mut timely::worker::Worker<A>,
+pub(super) fn construct<A: Scheduler + 'static, S: Scope<Timestamp = Timestamp>>(
```

Comment:
Nit: `A` seems like a strange name for a `Scheduler`. But `S` is already taken, so I don't have a better suggestion either 😅

Reply:
Solved! Scope is G, scheduler is S. At least we G for scope in many places, so it's not too bad.

Reply:
Works for me! Could you also change the name of the `DemuxState` type variable?